linkedlist/linkedList.go

```diff
@@ -68,34 +115,18 @@ func (l *LinkedList) Insert(index uint, val int) bool {
 	newNode.Next = current.Next
 	current.Next = newNode
 	l.length++
-	return true
-}
-
-func (l *LinkedList) Remove(index uint) bool {
-	if index >= l.length {
-		return false
-	}
-
-	if index == 0 {
-		l.head = l.head.Next
-		l.length--
-		return true
-	}
 
-	current := l.head
-	for i := uint(0); i < index-1; i++ {
-		if current.Next == nil {
-			return false
+	partIndex := int(index / part)
```

Comment:
In insert we have a condition that if the index is 0 it just change the node addresses and return
So in this design if i create 100 items with index 0 we have 100 nodes but 1 cached nodes to search for
So in Search by value we only spawn one goroutine that searches up to max of the part variable we sat
and if the value is beyond it it will never search for it